web/stripe-template/api/routes/webhook/POST-subscription.ts

```diff
@@ -2,26 +2,28 @@ import { RouteContext } from "gadget-server";
 import { getStripeWebhookEvent } from "../../stripe";
 import { objKeyConvert, destructure } from "../../utils";
 
-/**
- * Route handler for POST webhook/subscription
- *
- * @param { RouteContext } route context - see: https://docs.gadget.dev/guides/http-routes/route-configuration#route-context
- *
- */
+export type StripeWebhookEvent = {
+  type: string;
+  data: {
+    object: any;
+  };
+};
+
 export default async function route({
   request,
   reply,
   api,
   logger,
   connections,
-}) {
+}: RouteContext<{
+  Body: StripeWebhookEvent;
+}>) {
   let event = request.body;
 
   try {
     event = getStripeWebhookEvent({
-      logger,
       request: request,
-      endpointSecret: process.env.STRIPE_SUBSCRIPTION_WEBHOOK_SECRET,
+      endpointSecret: String(process.env.STRIPE_SUBSCRIPTION_WEBHOOK_SECRET),
     });
   } catch (err) {
     return await reply.status(400).send();
```

Comment:
When I cancel a stripe subscription we delete the subscription. Instead, we should keep the record and update the status field, then check to make sure the user has an active subscription

Reply:
I've changed the Stripe subscription logic to immediately end the subscription so the delete webhook will be fired